cloudinit/cmd/status.py

```diff
@@ -353,69 +346,92 @@ def _get_error_or_running_from_systemd(
                 # running. MainPID being set back to 0 means control of the
                 # service/unit has exited in this case and
                 # "the process is no longer around".
-                continue
-        if states["ActiveState"] == "failed" or states["SubState"] == "failed":
-            # We have an error
-            return UXAppStatus.ERROR
+                return False
+        elif (
+            states["ActiveState"] == "failed" or states["SubState"] == "failed"
+        ):
+            return True
         # If we made it here, our unit is enabled and it hasn't exited
         # normally or exited with failure, so it is still running.
-        return UXAppStatus.RUNNING
+        return False
     # All services exited normally or aren't enabled, so don't report
     # any particular status based on systemd.
-    return None
+    return False
 
 
-def get_status_details(
-    paths: Optional[Paths] = None, wait: bool = False
-) -> StatusDetails:
-    """Return a dict with status, details and errors.
+def is_running(status_file, result_file) -> bool:
+    """Return True if cloud-init is running."""
+    return os.path.exists(status_file) and not os.path.exists(result_file)
 
-    @param paths: An initialized cloudinit.helpers.paths object.
 
-    Values are obtained from parsing paths.run_dir/status.json.
+def get_running_status(
+    status_file, result_file, boot_status_code, latest_event
+) -> RunningStatus:
+    """Return the running status of cloud-init."""
+    if is_running(status_file, result_file):
```

Comment:
this logic normally should work, but I think it would be slightly safer to do `if boot_status_code in DISABLED_BOOT_CODES` before `is_running(status_file, result_file)`. This is because I think that `boot_status_code` is a higher confidence signal than the file existence. 

A system that experiences a cloud-init crash that leaves behind a status file but not a result file will return `RunningStatus.RUNNING` (non-systemd). While this makes sense, if cloud-init is subsequently disabled we will still report `RunningStatus.RUNNING`.